packages/strapi-admin/ee/admin/containers/Roles/CreatePage/Permissions/index.js

```diff
@@ -0,0 +1,38 @@
+import React, { forwardRef, memo, useImperativeHandle } from 'react';
+import PropTypes from 'prop-types';
+import { Tabs } from '../../../../../../admin/src/components/Roles';
+import { roleTabsLabel as TAB_LABELS } from '../../../../../../admin/src/utils';
+import ContentTypes from '../ContentTypes';
+import layout from '../temp/fakeData';
+
+const Permissions = forwardRef(({ layout }, ref) => {
+  useImperativeHandle(ref, () => {
```

Comment:
Is this necessary? Not a huge fan of imperative things 😛

Reply:
In this case yes, due to the architecture of the component we need to have a way to retrieve the permissions from this component in his parent. We could move the logic up, but I kept the old component architecture and will check when the refactoring is over if it's still necessary.